apollo-backend/src/test/java/io/logz/apollo/BlockerTest.java

```diff
@@ -156,6 +158,45 @@ public void testServiceBlocker() throws Exception {
         ModelsGenerator.createAndSubmitDeployment(apolloTestClient, environment, okService, okDeployableVersion);
     }
 
+    @Test
+    public void testEnvironmentsStackBlocker() throws Exception {
+        ApolloTestClient apolloTestClient = Common.signupAndLogin();
+        ApolloTestAdminClient apolloTestAdminClient = Common.getAndLoginApolloTestAdminClient();
+
+        Environment blockedEnvironment = ModelsGenerator.createAndSubmitEnvironment(apolloTestClient);
+        Environment okEnvironment = ModelsGenerator.createAndSubmitEnvironment(apolloTestClient);
+        Service service = ModelsGenerator.createAndSubmitService(apolloTestClient);
+        Stack blockedEnvironmentsStack = ModelsGenerator.createAndSubmitEnvironmentsStack(apolloTestClient, Arrays.asList(blockedEnvironment.getId()));
+        DeployableVersion deployableVersion = ModelsGenerator.createAndSubmitDeployableVersion(apolloTestClient, service);
+
+        createAndSubmitBlocker(apolloTestAdminClient, "unconditional", "{}", null, null, blockedEnvironmentsStack);
+
+        assertThatThrownBy(() -> ModelsGenerator.createAndSubmitDeployment(apolloTestClient, blockedEnvironment, service, deployableVersion)).isInstanceOf(Exception.class)
```

Comment:
There is no more specific exception here? its a bit too broad, and can let us introduce future bugs

Reply:
Unfortunately no